packages/apollo-language-server/src/diagnostics.ts

```diff
@@ -11,25 +13,43 @@ import { GraphQLDocument } from "./document";
 import { highlightNodeForNode } from "./utilities/graphql";
 import { rangeForASTNode } from "./utilities/source";
 
+import { TypeKind } from "apollo/lib/printer/ast";
 import { getValidationErrors } from "apollo/lib/validation";
 
-export function collectDiagnostics(
+/**
+ * Build an array of code diagnostics for all GraphQL operations in a document.
+ * This function explicitly ignores type definitions.
+ */
+export function collectOperationDiagnostics(
```

Comment:
Note that this won't just collect operation definitions, but also fragment definitions (as it should). I think the right term to describe both is 'executable definitions'. So maybe `collectExecutableDefinitionDiagnositics` (I realize that's quite the mouthful)?

Reply:
I'd rather be on the more verbose side if it provides clarity. Thanks for the clarity 👍